tests/sycl/test_barrier.sycl

```diff
@@ -0,0 +1,53 @@
+/**
+ * traccc library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include <CL/sycl.hpp>
+
+#include "../../device/sycl/src/utils/barrier.hpp"
+
+TEST(CUDABarrier, BarrierAnd) {
+    ::sycl::queue queue;
+    constexpr std::size_t n_bools = 4;
+
+    bool *out = malloc_shared<bool>(n_bools, queue);
+
+    queue.submit([&](cl::sycl::handler &h) {
+        h.parallel_for(cl::sycl::range<1>(128), [=](::sycl::nd_item<1> item) {
```

Comment:
```suggestion
        h.parallel_for<class BarrierOrTest>(cl::sycl::range<1>(128), [=](::sycl::nd_item<1> item) {
```

The "class-less" formalism is an Intel extension. It's a little more robust to specify a class on this call that would identify the kernel.